smdebug/core/tfevent/timeline_file_writer.py

```diff
@@ -306,13 +306,15 @@ def _should_rotate_now(self, now_in_us):
         file_size, diff_in_seconds, diff_in_hours = self._get_rotation_info(now_in_us)
         rotation_policy = self._profiler_config_parser.config.trace_file.rotation_policy
 
-        if diff_in_hours != 0:
-            return True
+        #        if diff_in_hours != 0:
+        #            return True
 
         if diff_in_seconds > rotation_policy.file_close_interval:
             return True
 
         if file_size > rotation_policy.file_max_size:
+            if now_in_us != self.last_event_end_time_in_us:
+                self.last_event_end_time_in_us = now_in_us
```

Comment:
Can we add a test for these changes?

Reply:
This line fixes the problem of overwriting the same filename where we basically lost data during detailed profiling. I have tested this with long running training jobs and here is the result that shows that without the fix merged timeline is missing events:
![image](https://user-images.githubusercontent.com/12165606/113190023-bba1b080-9210-11eb-9e92-c380f29e5174.png)


With bugfix:
![image](https://user-images.githubusercontent.com/12165606/113190044-c0fefb00-9210-11eb-8f1a-933d73aef4ae.png)

The new testcases that will be added to the CI ensure this part is tested.